### PR TITLE
Convert ZCL attributes from strings when writing

### PR DESCRIPTION
**Pull request by puddly**
When writing attributes via the ZCL cluster management dialog, strings aren't currently converted into their correct type. I believe this is a regression.

We'll need to figure out a generic way to do string conversion for complex types, like arrays and structs.

### CODE REVIEW

**Comment by Monofin** [comment]
Hi there - what's needed to push this one forward, as I'm struggling with broken device settings that would be resolved by this? Happy to help out if there's anything I can do :-)

**Comment by FLab-Projects** [comment]
Hi! I am also having problems changing device settings. I have Tuya TZE204_qasjif9e and I can't change its parameters. Any news on enabling your fix?

Write if I can help with testing :)

**Comment by puddly** [comment]
I've not had time to write unit tests for this change so if you'd like to see it merged, 100% test coverage would be nice 😄

**Comment by Monofin** [comment]
Love to help, but I’m not that familiar with the unit test system in use - if you’ve got a hint as to where to get started I’ll have a look though! :-)

**Comment by javicalle** [comment]
> Love to help, but I’m not that familiar with the unit test system in use - if you’ve got a hint as to where to get started I’ll have a look though! :-)

Maybe you can look at the current tests:
* https://github.com/zigpy/zha/blob/dev/tests/test_application_helpers.py

File: zha/application/helpers.py [hunk 1]
```diff
@@ -126,6 +128,53 @@ async def get_matched_clusters(
...
+    """Convert user input to ZCL value."""
+    if issubclass(field_type, enum.Flag):
+        if isinstance(value, str):
+            with contextlib.suppress(ValueError):
+                value = int(value)
+
+        if isinstance(value, int):
+            value = field_type(value)
+        elif isinstance(value, str):
+            # List of flags: `SomeFlag.field1 | field2
```
Comment (TheJulianJES) [inline zha/application/helpers.py]: Remove ` or add another one?

File: zha/application/helpers.py [hunk 2]
```diff
@@ -134,32 +183,17 @@ def convert_to_zcl_values(
...
-                else field.type(value)
-            )
-        else:
-            value = field.type(value)
         _LOGGER.debug(
             "Converted ZCL schema field(%s) value from: %s to: %s",
             field.name,
-            fields[field.name],
             value,
+            fields[field.name],
```
Comment (TheJulianJES) [inline zha/application/helpers.py]: Should this be `converted_fields[field.name]`?

**Review by TheJulianJES — APPROVED** [review]

**Pull request merged**